src/main/java/com/avispl/symphony/dal/communicator/management/zoom/rooms/ZoomRoomsCommunicator.java

```diff
@@ -0,0 +1,395 @@
+
+        Map<String, String> systemUnitParameters = parseZoomRoomsProperties(systemUnit, "*s SystemUnit");
+
+        lineParameters.forEach((key, value) -> {
+            if(key.endsWith("Name") || key.endsWith(" Selected")){
+                if(key.startsWith("Audio")){
+                    statistics.put("Audio Settings#" + key, value);
+                } else if (key.startsWith("Video Camera")){
+                    statistics.put("Video Camera Settings#" + key, value);
+                }
+            }
+        });
+
+        systemUnitParameters.forEach((key, value) -> {
+            String keyValue = key.replace("SystemUnit ", "").replace("room_info", "").trim();
+            switch (keyValue){
+                case "room_version":
+                    statistics.put("Zoom Rooms Version", value);
+                    break;
+                case "meeting_number":
+                    statistics.put("Meeting Number", value);
+                    break;
+                case "account_email":
+                    statistics.put("Account Email", value);
+                    break;
+                case "room_name":
+                    statistics.put("Room Name", value);
+                    break;
+                case "platform":
+                    statistics.put("Platform", value);
+                    break;
+            }
+        });
+    }
+
+    /**
+     * Parse properties parameters based on the provided start of the line,
+     * so the expected format of data provided here is:
+     *      Video Camera Line 1 Name: Camera Name
+     *      .....
+     *      Video Camera Line 2 Name: Camera Name
+     *
+     * @param response string from Zoom Rooms CLI
+     * @param propertyIdentifier supposed start of the string
+     * @return map of a format <Video Camera Line 1 Name: Camera Name, Video Camera Line 2 Name: Camera Name....>
+     */
+    private Map<String, String> parseZoomRoomsProperties(String response, String propertyIdentifier){
+        Map<String, String> parameters = new HashMap<>();
+        Arrays.asList(response.split("\n")).forEach(s -> {
+            if(s.startsWith(propertyIdentifier)){
+                String[] values = s.split(":");
+                parameters.put(values[0].replace("*s ", ""), values[1].replace("\r", ""));
+            }
+        });
+        return parameters;
+    }
+
+    /**
+     * Join meeting that is currently in progress or start a scheduled one.
+     * @param value - contains dialString of a format %meetingNumber%.%meetingPassword%@zoomcrc.com
+     *                password is optional.
+     * @return meetingId if operation is successful
+     */
+    private String joinStartMeeting(String value) throws Exception {
+        boolean hasPassword;
+        String meetingNumber;
+        String meetingPassword;
+
+        Pattern dialStringPattern = Pattern.compile("(\\d+)\\.?(\\d+)?(@zoomcrc\\.com)");

```

Comment:
Ok, I see. Do you think it should be less specific in terms of domain, like if "any domain" is there? Originally I had domain part set as optional part of the dial string, which would make it possible to dial the meeting with both _meetingId.meetingPassword@domain.xxx_ and just _meetingId.meetingPassword_. This would also make it a bit less tedious to dial zoom meeting from the device, so domain part could've been just omitted, otherwise, for scheduled meetings, for instance, it would be "any domain" or "a domain that contains zoom keyword in it". Should I follow this approach instead?

Reply:
@MaksimR-AVISPL I do not have enough information to provide an advice here. In order to do so, I need answers to questions which I'm asking for some time now. Can Zoom room connect to other SIP device and what would be the command for that and dial address syntax? Can Zoom room connect to other meeting (not Zoom) as long as it is reachable by SIP (CMS, Pexip, RMX, whatever). If not - what would actually happen if you try to dial to another SIP address (and not Zoom meeting).
The problem is that you writing generic codec adapter which Symphony will use for any kind of connections. There is no code in Symphony saying to use this adapter for Zoom meetings only. But you are coding this adapter with assumption that it will only be used for Zoom meetings! Only when you have answers to all the questions (including the question about whether all Zoom meetings will use the same domain) you can write proper code for dial method - validate address, add proper restrictions and error handling there. I can only provide advice when I have clear picture of what is supported and what exactly will happen in each case.